docs/keepalived.md

```diff
@@ -0,0 +1,72 @@
+* run `sudo ip a`
+* you should see the VIP (in example above, VIP is set as 10.234.250.201)
+
+```
+2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP group default qlen 1000
+    link/ether 06:c9:f6:00:0a:38 brd ff:ff:ff:ff:ff:ff
+    inet 10.234.250.199/26 brd 10.234.250.255 scope global eth0
+       valid_lft forever preferred_lft forever
+    inet 10.234.250.201/32 scope global eth0
+       valid_lft forever preferred_lft forever
+```
+* The VIP is up, you can perform further testing and access your backend services using the VIP
+
+## Failover scenario
+* Let s stop haproxy on first node by running `monit stop haproxy`
+* Let s run `ip a` on first node
+```
+2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP group default qlen 1000
+    link/ether 06:c9:f6:00:0a:38 brd ff:ff:ff:ff:ff:ff
+    inet 10.234.250.199/26 brd 10.234.250.255 scope global eth0
+       valid_lft forever preferred_lft forever
+```
+* no more VIP, let s look at our second node
+```
+2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP group default qlen 1000
+    link/ether 06:38:ce:00:0a:39 brd ff:ff:ff:ff:ff:ff
+    inet 10.234.250.200/26 brd 10.234.250.255 scope global eth0
+       valid_lft forever preferred_lft forever
+    inet 10.234.250.201/32 scope global eth0
+       valid_lft forever preferred_lft forever
+```
+* It works ! If we look at logs on first node :
+```
+Dec  7 12:47:34 localhost Keepalived_vrrp[4558]: VRRP_Script(check_haproxy) failed
+Dec  7 12:47:34 localhost Keepalived_vrrp[4558]: VRRP_Instance(haproxy_keepalived_mysql_infra_check_haproxy) Effective priority = 101
+Dec  7 12:47:35 localhost Keepalived_vrrp[4558]: VRRP_Instance(haproxy_keepalived_mysql_infra_check_haproxy) Received higher prio advert 102
+Dec  7 12:47:35 localhost Keepalived_vrrp[4558]: VRRP_Instance(haproxy_keepalived_mysql_infra_check_haproxy) Entering BACKUP STATE
+```
+and second node :
+```
+Dec  7 12:47:35 localhost Keepalived_vrrp[4544]: VRRP_Instance(haproxy_keepalived_mysql_infra_check_haproxy) forcing a new MASTER election
+Dec  7 12:47:36 localhost Keepalived_vrrp[4544]: VRRP_Instance(haproxy_keepalived_mysql_infra_check_haproxy) Transition to MASTER STATE
+Dec  7 12:47:37 localhost Keepalived_vrrp[4544]: VRRP_Instance(haproxy_keepalived_mysql_infra_check_haproxy) Entering MASTER STATE
+```
+* Same scenario if you stop the master node :
+```
+Dec  7 12:55:52 localhost Keepalived_vrrp[4544]: VRRP_Instance(haproxy_keepalived_mysql_infra_check_haproxy) Received higher prio advert 103
+Dec  7 12:55:52 localhost Keepalived_vrrp[4544]: VRRP_Instance(haproxy_keepalived_mysql_infra_check_haproxy) Entering BACKUP STATE
+Dec  7 12:58:22 localhost Keepalived_vrrp[4544]: VRRP_Instance(haproxy_keepalived_mysql_infra_check_haproxy) Transition to MASTER STATE
+Dec  7 12:58:23 localhost Keepalived_vrrp[4544]: VRRP_Instance(haproxy_keepalived_mysql_infra_check_haproxy) Entering MASTER STATE
+```
+
```

Comment:
Missing some test cases (at least mention they need to be performed)
* kill haproxy VM (using underlying Iaas) and check keepalived log extracts. 
* Test the bosh update scenario (stemcell ugrade/version upgrade/haproxy properties upgrade)